studio/components/SalaryCsvInput/SalariesInput.tsx

```diff
@@ -0,0 +1,146 @@
+import { set, StringInputProps } from "sanity";
+import { Box, Grid, Inline, Stack, Text, useTheme, useToast } from "@sanity/ui";
+import { ChangeEvent } from "react";
+import styles from "./salariesInput.module.css";
+import { UploadIcon } from "@sanity/icons";
+import {
+  salariesAsStoredString,
+  salariesFromCsvString,
+  salariesFromStoredString,
+} from "./salariesParseUtils";
+import { SalariesParseErrorsToastDescription } from "./SalariesParseErrorsToastDescription";
+import SalaryNumberInput from "./SalaryNumberInput";
+
+const UPLOAD_CSV_INPUT_ID = "upload-csv-input";
+
+export const SalariesInput = (props: StringInputProps) => {
+  const toast = useToast();
+  const theme = useTheme();
+  const prefersDark = theme.sanity.v2?.color._dark ?? false;
+
+  const salaries =
+    props.value === undefined
+      ? undefined
+      : salariesFromStoredString(props.value);
+
+  async function handleFileRead(e: ProgressEvent<FileReader>): Promise<void> {
+    const fileData = e.target?.result;
+    if (fileData === null || typeof fileData !== "string") {
+      toast.push({
+        title: "Invalid data",
+        description: "Verify the file content and try again.",
+        status: "error",
+      });
+      return;
+    }
+    const salariesParseResult = salariesFromCsvString(fileData);
+    if (!salariesParseResult.ok) {
+      toast.push({
+        title: "Invalid salaries data",
+        description: (
+          <SalariesParseErrorsToastDescription
+            errors={salariesParseResult.error}
+          />
+        ),
+        status: "error",
+      });
+      return;
+    }
+    props.onChange(set(salariesAsStoredString(salariesParseResult.value)));
+  }
+
+  function handleFileChange(e: ChangeEvent<HTMLInputElement>): void {
+    if (e.target.files !== null && e.target.files.length > 0) {
+      const file = e.target.files[0];
+      if (file.type === "text/csv") {
+        const reader = new FileReader();
+        reader.onload = handleFileRead;
+        reader.readAsText(file);
+      } else {
+        toast.push({
+          title: "Invalid salaries file type",
+          description: "Verify the file extension and try again.",
+          status: "error",
+        });
+      }
+    }
+    // reset to allow subsequent uploads of the same file
+    e.target.value = "";
+  }
+
+  function handleYearSalaryChange(year: string, salary: number): void {
+    props.onChange(
+      set(
+        salariesAsStoredString({
+          ...salaries,
+          [year]: salary,
+        }),
+      ),
+    );
+  }
+
+  return (
+    <Stack space={4} className={prefersDark ? styles.darkTheme : ""}>
+      <Inline space={2}>
+        <div>
+          {/*https://developer.mozilla.org/en-US/docs/Web/API/File_API/Using_files_from_web_applications#using_a_label_element_to_trigger_a_hidden_file_input_element*/}
```

Comment:
```suggestion

```

Reply:
Is this not a useful reference? Added some more context now to clarify the purpose.